src/sinks/redis/config.rs

```diff
@@ -0,0 +1,162 @@
+use redis::{aio::ConnectionManager, RedisResult};
+use snafu::prelude::*;
+
+use crate::sinks::prelude::*;
+
+use super::{sink::RedisSink, RedisCreateFailedSnafu};
+
+/// Redis data type to store messages in.
+#[configurable_component]
+#[derive(Clone, Copy, Debug, Derivative)]
+#[derivative(Default)]
+#[serde(rename_all = "lowercase")]
+pub enum DataTypeConfig {
+    /// The Redis `list` type.
+    ///
+    /// This resembles a deque, where messages can be popped and pushed from either end.
+    ///
+    /// This is the default.
+    #[derivative(Default)]
+    List,
+
+    /// The Redis `channel` type.
+    ///
+    /// Redis channels function in a pub/sub fashion, allowing many-to-many broadcasting and receiving.
+    Channel,
+}
+
+/// List-specific options.
+#[configurable_component]
+#[derive(Clone, Copy, Debug, Derivative, Eq, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub struct ListOption {
+    /// The method to use for pushing messages into a `list`.
+    pub(super) method: Method,
+}
```

Comment:
This is a `configurable_component.` https://vector.dev/docs/reference/configuration/sinks/redis/#list_option 

I guess it's been done this way so that we can configure `list_option.method: lpush`.

Changing this would be a breaking change, so I don't think we should..